lib/ansible/modules/lineinfile.py

```diff
@@ -312,7 +354,18 @@ def present(module, dest, regexp, line, insertafter, insertbefore, create,
                 if firstmatch:
                     break
 
-    # 2. When no match found on the previous step,
+    # 2. Second check that there is no match for literal:
+    if literal is not None:
+        for lineno, b_cur_line in enumerate(b_lines):
+            match_found = to_bytes(
+                literal, errors='surrogate_or_strict') in b_cur_line
```

Comment:
No reason to break this line.
```suggestion
            match_found = to_bytes(literal, errors='surrogate_or_strict') in b_cur_line
```